packages/venia-ui/lib/components/Image/image.js

```diff
@@ -21,72 +38,142 @@ const Image = props => {
         onError,
         onLoad,
         placeholder,
+        resource,
+        resourceHeight,
+        resourceWidth,
+        sizes,
         src,
-        fileSrc,
+        type,
+        usePlaceholder,
         ...rest
     } = props;
 
     const talonProps = useImage({
         onError,
-        onLoad,
-        placeholder
+        onLoad
     });
 
     const {
         handleError,
         handleImageLoad,
-        hasError,
         isLoaded,
         shouldRenderPlaceholder
     } = talonProps;
 
     const classes = mergeClasses(defaultClasses, propsClasses);
 
-    // Render a placeholder until the image is loaded.
-    const placeholderImage = shouldRenderPlaceholder ? (
-        <img className={classes.root} src={placeholder} alt={alt} {...rest} />
-    ) : null;
-
-    const imageClass =
-        classes.root + ' ' + (isLoaded ? classes.loaded : classes.notLoaded);
-
-    const imageSrcset = useMemo(
-        () => generateSrcset(fileSrc, 'image-product'),
-        [fileSrc]
+    // A placeholder to use until the image is loaded.
+    // This is used both for user experience and layout purposes.
+    // Callers can disable the "user experience" part by setting usePlaceholder to false.
+    let placeholderClass;
+    if (!usePlaceholder) {
+        placeholderClass = classes.placeholder_layoutOnly;
+    } else {
+        placeholderClass = shouldRenderPlaceholder
+            ? classes.placeholder
+            : classes.placeholder_layoutOnly;
+    }
+    const placeholderImageClass = `${classes.image} ${placeholderClass}`;
+    const placeholderImage = (
+        <img
+            alt={alt}
+            className={placeholderImageClass}
+            loading="eager"
+            src={placeholder}
+            {...rest}
+        />
     );
 
-    const actualImage = !hasError && (
+    /*
+     * These don't live in the talon because they depend on @magento/venia-drivers.
```

Comment:
Can you clarify this comment? What do you mean by "these"?

Reply:
Clarified in https://github.com/magento/pwa-studio/pull/1871/commits/8e815492172cac77557241c4e7dc51a15a634220.